c-dependencies/js-compute-runtime/xqd.wit

```diff
@@ -0,0 +1,510 @@
+world xqd-world {
+  import fastly: interface {
+    // TODO: split this up into function-specific error enums
+    enum error {
+      /// Unknown error value.
+      /// It should be an internal error if this is returned.
+      unknown-error,
+      /// Generic error value.
+      /// This means that some unexpected error occurred during a hostcall.
+      generic-error,
+      /// Invalid argument.
+      invalid-argument,
+      /// Invalid handle.
+      /// Thrown when a handle is not valid. E.G. No dictionary exists with the given name.
+      bad-handle,
+      /// Buffer length error.
+      /// Thrown when a buffer is too long.
+      buffer-len,
+      /// Unsupported operation error.
+      /// This error is thrown when some operation cannot be performed, because it is not supported.
+      unsupported,
+      /// Alignment error.
+      /// This is thrown when a pointer does not point to a properly aligned slice of memory.
+      bad-align,
+      /// Invalid HTTP error.
+      /// This can be thrown when a method, URI, header, or status is not valid. This can also
+      /// be thrown if a message head is too large.
+      http-invalid,
+      /// HTTP user error.
+      /// This is thrown in cases where user code caused an HTTP error. For example, attempt to send
+      /// a 1xx response code, or a request with a non-absolute URI. This can also be caused by
+      /// an unexpected header: both `content-length` and `transfer-encoding`, for example.
+      http-user,
+      /// HTTP incomplete message error.
+      /// This can be thrown when a stream ended unexpectedly.
+      http-incomplete,
+      /// A `None` error.
+      /// This status code is used to indicate when an optional value did not exist, as opposed to
+      /// an empty value.
+      /// Note, this value should no longer be used, as we have explicit optional types now.
+      optional-none,
+      /// Message head too large.
+      http-head-too-large,
+      /// Invalid HTTP status.
+      http-invalid-status,
```

Comment:
This is missing the newer error case for exceeding limits (https://github.com/fastly/Viceroy/commit/8ee547b53cadb8366082bd7ca931be0a3a770411):
```suggestion
      http-invalid-status,
      // Limit exceeded
      // This is returned when an attempt to allocate a resource has exceeded the maximum number of 
      // resources permitted. For example, creating too many response handles.
      limit-exceeded,
```

Reply:
Sure, added and also extended a new JS error message for this as well.